### ISSUE TITLE
i18next adding Translations using the API arrays

### ISSUE THREAD

**Issue by MrKinau** [WeblateOrg/weblate#6363]
**Describe the issue**

After adding Translations using the APIs [file uploading function](https://docs.weblate.org/en/latest/api.html#post--api-translations-(string-project)-(string-component)-(string-language)-file-) to the source language, they do not seem to appear on any other languages (the count does not update), but after clicking on "all strings" of another language you see the keys and they have set the source language values set. They do not appear in the output JSON, if you download it using the [download function](https://docs.weblate.org/en/latest/api.html#post--api-translations-(string-project)-(string-component)-(string-language)-file-), unless you change it's values in the language, but they seem to randomly disappear again even if you edited the value.

If I use the [add translations function](https://docs.weblate.org/en/latest/api.html#post--api-translations-(string-project)-(string-component)-(string-language)-file-). Firstly the API documentation is incorrect, it requires an array as a value (at least for i18next), not a string and it seems to work for normal keys, but I am not sure how to correctly add Arrays using this method. Do I need to send a request for each array value in the format:
```json
{"key": "my.translation.key[0]", "value": ["Test Value 1"]}
```
If I do so it looks fine in the panel, but if I start translating the array, some values randomly disappear in the file I download, although they're shown in the Weblate interface and can be translated correctly. Not sure in which conditions they disappear.

**I already tried**

I've tried both methods I mentioned above and the last one seems to be working for normal strings, but for arrays it feels wrong to me to add arrays this way and it's quite random if it works or not.

- [X] I've read and searched [the docs](https://docs.weblate.org/) and did not find the answer there.
      If you didn’t try already, try to search there what you wrote above.

**To Reproduce the issue**

File Upload:

1. Create a new project and one component with i18next format
2. Add two languages to the component
3. Use the APIs [file uploading function](https://docs.weblate.org/en/latest/api.html#post--api-translations-(string-project)-(string-component)-(string-language)-file-) to upload a file for the source language adding translations
4. Click on the target language and see no strings, but if you click on the "0 - All string", you see the strings translated in the source language

Add Translations:

1. Create a new project and one component with i18next format
2. Add two languages to the component
3. Use the APIs [add translations function](https://docs.weblate.org/en/latest/api.html#post--api-translations-(string-project)-(string-component)-(string-language)-file-) to add translations as array to the source language. Tested with this array:
```json
{
    "test": {
        "key": [
            "§r",
            "§7Rank: §6{{rank}}",
            "§7Gold: §6{{money}}",
            "§7Level: §a{{level}}",
            "§r",
            "§8» §7Manage"
        ]
    }
}
```
4. Click on the target language and translate the new strings (Some strings are the same in input and output language) and download the file via the [download function](https://docs.weblate.org/en/latest/api.html#post--api-translations-(string-project)-(string-component)-(string-language)-file-) and see some strings missing in the target language.

**Expected behavior**

File Upload:

If I use the [file uploading function](https://docs.weblate.org/en/latest/api.html#post--api-translations-(string-project)-(string-component)-(string-language)-file-) I would expect the keys to be added to all languages correctly if I add them to the source language.

File Upload:

If I use the [add translations function](https://docs.weblate.org/en/latest/api.html#post--api-translations-(string-project)-(string-component)-(string-language)-file-) I would expect to see the translated keys in the downloaded file, after I translated them.

**Screenshots**

If required I can add screens or video.

**Server configuration and status**

Weblate installation: Docker

 * Weblate: 4.7.1
 * Django: 3.2.4
 * siphashc: 2.1
 * translate-toolkit: 3.3.6
 * lxml: 4.6.3
 * Pillow: 8.2.0
 * bleach: 3.3.0
 * python-dateutil: 2.8.1
 * social-auth-core: 4.1.0
 * social-auth-app-django: 4.0.0
 * django-crispy-forms: 1.12.0
 * oauthlib: 3.1.1
 * django-compressor: 2.4.1
 * djangorestframework: 3.12.4
 * django-filter: 2.4.0
 * django-appconf: 1.0.4
 * user-agents: 2.2.0
 * filelock: 3.0.12
 * setuptools: 40.8.0
 * jellyfish: 0.8.2
 * openpyxl: 3.0.7
 * celery: 5.1.2
 * kombu: 5.1.0
 * translation-finder: 2.9
 * weblate-language-data: 2021.5
 * html2text: 2020.1.16
 * pycairo: 1.16.2
 * pygobject: 3.30.4
 * diff-match-patch: 20200713
 * requests: 2.25.1
 * django-redis: 5.0.0
 * hiredis: 2.0.0
 * sentry_sdk: 1.1.0
 * Cython: 0.29.23
 * misaka: 2.1.1
 * GitPython: 3.1.18
 * borgbackup: 1.1.16
 * pyparsing: 2.4.7
 * pyahocorasick: 1.4.2
 * python-redis-lock: 3.7.0
 * Python: 3.7.3
 * Git: 2.20.1
 * psycopg2: 2.9.1
 * psycopg2-binary: 2.9.1
 * phply: 1.2.5
 * chardet: 4.0.0
 * ruamel.yaml: 0.17.10
 * tesserocr: 2.5.2
 * akismet: 1.1
 * boto3: 1.17.102
 * zeep: 4.0.0
 * aeidon: 1.9
 * iniparse: 0.5
 * mysqlclient: 2.0.3
 * Mercurial: 5.8
 * git-svn: 2.20.1
 * git-review: 2.1.0
 * Redis server: 6.2.4
 * PostgreSQL server: 13.3
 * Database backends: django.db.backends.postgresql
 * Cache backends: default:RedisCache, avatar:FileBasedCache
 * Email setup: django.core.mail.backends.smtp.EmailBackend: pro1.mail.ovh.net
 * OS encoding: filesystem=utf-8, default=utf-8
 * Celery: redis://cache:6379/1, redis://cache:6379/1, regular
 * Platform: Linux 5.4.44-2-pve (x86_64)

**Weblate deploy checks**

```
System check identified some issues:
WARNINGS:
?: (security.W004) You have not set a value for the SECURE_HSTS_SECONDS setting. If your entire site is served only over SSL, you may want to consider setting a value and enabling HTTP Strict Transport Security. Be sure to read the documentation first; enabling HSTS carelessly can cause serious, irreversible problems.
?: (security.W008) Your SECURE_SSL_REDIRECT setting is not set to True. Unless your site should be available over both SSL and non-SSL connections, you may want to either set this setting True or configure a load balancer or reverse-proxy server to redirect all connections to HTTPS.
?: (security.W012) SESSION_COOKIE_SECURE is not set to True. Using a secure-only session cookie makes it more difficult for network traffic sniffers to hijack user sessions.
?: (security.W018) You should not have DEBUG set to True in deployment.

INFOS:
?: (weblate.I021) Error collection is not set up, it is highly recommended for production use
        HINT: https://docs.weblate.org/en/weblate-4.7.1/admin/install.html#collecting-errors
?: (weblate.I028) Backups are not configured, it is highly recommended for production use
        HINT: https://docs.weblate.org/en/weblate-4.7.1/admin/backup.html
?: (weblate.I031) New Weblate version is available, please upgrade to 4.7.2.
        HINT: https://docs.weblate.org/en/weblate-4.7.1/admin/upgrade.html

System check identified 7 issues (1 silenced).
**Additional context**
```

**Comment by nijel** [comment]
> Firstly the API documentation is incorrect, it requires an array as a value (at least for i18next)

Thanks, I've fixed the docs.

> Click on the target language and see no strings, but if you click on the "0 - All string", you see the strings translated in the source language

This could be that the component is still being updated, the number should update once it is fully processed.

>  see some strings missing in the target language.

Can you please check whether this happens only when some strings in the array are not translated?

**Comment by MrKinau** [comment]
> This could be that the component is still being updated, the number should update once it is fully processed.

Should it update automatically, because it does not update automatically, but the count isn't that of a big deal. The real problem is that:
1. The strings appear in the source language instead of appearing untranslated.
2. You need to edit and save the strings (even if source and target language are equal) to add them to the json.

> Can you please check whether this happens only when some strings in the array are not translated?

As I mentioned it appeared to happen quite random. I've tested a bit around and I can' reproduce this anymore. I think my main issue was, that I sent all add translation requests at the same time and they have not arrived in the correct order and some keys didn't appear even in the source language, but thats my fault.

**Comment by nijel** [comment]
> This could be that the component is still being updated, the number should update once it is fully processed.

Okay, I was wrong, the counts are not properly updated in this case. Fixed in 7950329034.

> The strings appear in the source language instead of appearing untranslated.

Can you please describe your setup? On which language do you add the string? Is that the source langauge? And where do you observe this behavior?

> You need to edit and save the strings (even if source and target language are equal) to add them to the json.

This is currently expected, only translated keys are stored in the JSON file.

**Comment by MrKinau** [comment]
> Can you please describe your setup? On which language do you add the string? Is that the source langauge? And where do you observe this behavior?

I've configured German as source language and English one target language and updated the german language file using the API to add a string.
